code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -197,22 +199,23 @@ func (sc *StorageChallenge) Save(state cstate.StateContextI, scAddress string) e
 type ValidationNode struct {
 	ID                string             `json:"id"`
 	BaseURL           string             `json:"url"`
+	ProviderType      spenum.Provider    `json:"provider_type"`
 	PublicKey         string             `json:"-" msg:"-"`
 	StakePoolSettings stakepool.Settings `json:"stake_pool_settings"`
 }
 
 // validate the validator configurations
-func (sn *ValidationNode) validate(conf *Config) (err error) {
+func (sn *ValidationNode) validate(_ *Config) (err error) {
 	if strings.Contains(sn.BaseURL, "localhost") &&
 		node.Self.Host != "localhost" {
-		return errors.New("invalid validator base url")
+		return errors.New("invalid vali1dator base url")
 	}
 
 	return
 }
 
-func (sn *ValidationNode) GetKey(globalKey string) datastore.Key {
-	return datastore.Key(globalKey + "validator:" + sn.ID)
+func (sn *ValidationNode) GetKey(_ string) datastore.Key {
+	return provider.GetKey(sn.ID)
```

Comment:
Please do not remove the globalKey(SC address), it helps to differ the keys between SCs. Removing it increase the possibility to get conflict keys.

Reply:
The whole point is to increase the chance of conflicting keys, so we can detect different providers with the same id.